core/dnsserver/onstartup_test.go

```diff
@@ -0,0 +1,12 @@
+package dnsserver
+
+import "testing"
+
+func TestCheckDomainName(t *testing.T) {
+	if !checkDomainName("an-example.com") {
```

Comment:
Won’t this fail with your latest commit? There is no trailing dot in the tested input.

Reply:
I think, this will fail, I need to update test too.